### PR TITLE
[GCS] Add StatsInfoAccessor to GCS Client

### PR DESCRIPTION
**Pull request by micafan**
## Why are these changes needed?

This class `StatsInfoAccessor` includes all the methods that are related to accessing
 stats in the GCS.

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: src/ray/core_worker/profiling.cc [hunk 1]
```diff
@@ -18,31 +18,35 @@ Profiler::Profiler(WorkerContext &worker_context, const std::string &node_ip_add
...
+    cur_profile_data->set_node_ip_address(rpc_profile_data_->node_ip_address());
+
+    rpc_profile_data_.swap(cur_profile_data);
+
+    if (!gcs_client_->Stats().AsyncAddProfileData(cur_profile_data, nullptr).ok()) {
       RAY_LOG(WARNING) << "Failed to push profile events to GCS.";
     } else {
-      RAY_LOG(DEBUG) << "Pushed " << rpc_profile_data_.profile_events_size()
+      RAY_LOG(DEBUG) << "Pushed " << cur_profile_data->profile_events_size()
                      << "events to GCS.";
```
Comment (ffbin) [inline src/ray/core_worker/profiling.cc]: << " events to GCS.";

Comment (micafan) [reply src/ray/core_worker/profiling.cc]: done

File: src/ray/core_worker/profiling.h [hunk 2]
```diff
@@ -18,14 +18,11 @@ class Profiler {
...
   // Add an event to the queue to be flushed periodically.
-  void AddEvent(const rpc::ProfileTableData::ProfileEvent &event) LOCKS_EXCLUDED(mutex_);
+  void AddEvent(const rpc::ProfileTableData::ProfileEvent &event);
 
  private:
   // Flush all of the events that have been added since last flush to the GCS.
-  void FlushEvents() LOCKS_EXCLUDED(mutex_);
-
-  // Mutex guarding rpc_profile_data_.
-  absl::Mutex mutex_;
```
Comment (zhijunfu) [inline src/ray/core_worker/profiling.h]: Would you mind explain on why this mutext is no longer needed?

Comment (micafan) [reply src/ray/core_worker/profiling.h]: Actually AddEvent is still need this lock. I will fix this.

File: src/ray/object_manager/object_manager.cc [hunk 3]
```diff
@@ -804,15 +804,16 @@ std::shared_ptr<rpc::ObjectManagerClient> ObjectManager::GetRpcClient(
...
-  rpc::ProfileTableData profile_info;
-  profile_info.set_component_type("object_manager");
-  profile_info.set_component_id(self_node_id_.Binary());
+std::shared_ptr<rpc::ProfileTableData> ObjectManager::GetAndResetProfilingInfo() {
+  auto profile_info = std::make_shared<rpc::ProfileTableData>();
+  profile_info->set_component_type("object_manager");
+  profile_info->set_component_id(self_node_id_.Binary());
 
   {
+    // TODO(micafan) Remove this mutex in next PR.
```
Comment (raulchen) [inline src/ray/object_manager/object_manager.cc]: how do you plan to remove this mutex?

Comment (micafan) [reply src/ray/object_manager/object_manager.cc]: Not a big issue, remove the comments already.

**Review by ffbin — APPROVED** [review]
LGTM

**Review by raulchen — APPROVED** [review]

**Pull request merged**